src/main/java/uk/gov/ons/census/casesvc/service/NewAddressReportedService.java

```diff
@@ -278,6 +304,84 @@ private CaseMetadata metadataFromSourceCase(CaseMetadata sourceMetadata) {
+    newCase.setUprn(dummyUprn);
+  }
+
+  private void sendNewAddressToAims(ResponseManagementEvent newAddressEvent, String dummyUprn) {
+    ResponseManagementEvent enhancedNewAddressEvent =
+        buildEnhancedNewAddressEvent(newAddressEvent, dummyUprn);
+
+    ListenableFuture<String> future =
+        pubSubTemplate.publish(aimsNewAddressTopic, enhancedNewAddressEvent);
+    try {
+      future.get(publishTimeout, TimeUnit.SECONDS);
+    } catch (InterruptedException | ExecutionException | TimeoutException e) {
+      throw new RuntimeException(e);
+    }
+
+    // Warn about dummy UPRN sent to AIMS out of sync with RM if the transaction rolls back.
+    // There's nothing we can do about these more elegantly - PubSub needs to become transactional.
+    if (TransactionSynchronizationManager.isSynchronizationActive()) {
+      TransactionSynchronizationManager.registerSynchronization(
+          new TransactionSynchronization() {
+            @Override
+            public void afterCompletion(int status) {
+              if (status == STATUS_ROLLED_BACK) {
+                // All we can do is log. Hopefully this is enough info to manually patch the data
+                // in AIMS to get rid of any duplicate addresses
+                log.with("dummy_uprn", dummyUprn)
+                    .with(
+                        "case_id",
+                        newAddressEvent.getPayload().getNewAddress().getCollectionCase().getId())
+                    .error("Transaction rolled back after PubSub message sent");
+              }
+            }
+          });
+    }
+  }
+
+  private ResponseManagementEvent buildEnhancedNewAddressEvent(
+      ResponseManagementEvent newAddressEvent, String dummyUprn) {
+    EventDTO event = new EventDTO();
+    event.setChannel("RM");
+    event.setType(NEW_ADDRESS_ENHANCED);
+    event.setTransactionId(newAddressEvent.getEvent().getTransactionId());
+    event.setDateTime(OffsetDateTime.now());
+    event.setSource("CASE_PROCESSOR");
+
+    CollectionCase sourceCollectionCase =
+        newAddressEvent.getPayload().getNewAddress().getCollectionCase();
+
+    Address address = new Address();
+    address.setAddressLine1(sourceCollectionCase.getAddress().getAddressLine1());
+    address.setAddressLine2(sourceCollectionCase.getAddress().getAddressLine2());
+    address.setAddressLine3(sourceCollectionCase.getAddress().getAddressLine3());
+    address.setTownName(sourceCollectionCase.getAddress().getTownName());
+    address.setPostcode(sourceCollectionCase.getAddress().getPostcode());
+    address.setAddressType(sourceCollectionCase.getAddress().getAddressType());
+    address.setAddressLevel(sourceCollectionCase.getAddress().getAddressLevel());
+    address.setAddressLevel(sourceCollectionCase.getAddress().getAddressLevel());
+    address.setLatitude(sourceCollectionCase.getAddress().getLatitude());
+    address.setLongitude(sourceCollectionCase.getAddress().getLongitude());
```

Comment:
question about where the lat/long would even come from... I'm not sure RM should be telling AIMS this, or whether it's AIMS' job to actually flesh out this detail - if you copied this verbatim from my branch, it might be wrong

Reply:
It's in the confluence doc, in fact it turns out we should be setting the survey too

Reply:
the confluence doc has been wrong many times. AIMS definitely don't need to know the survey -see my comment in confluence